File: 70-ClimbingStairs/70-ClimbingStairs.py
# Last updated: 31/07/2025, 21:19:08
class Solution:
    def climbStairs(self, n: int) -> int:
        r=0;
        sumf=0;
        while(n>=r):
            sumf=sumf+math.comb(n,r)
            n=n-1
            r=r+1
        return sumf

File: 70-ClimbingStairs/70-ClimbingStairs.java
// Last updated: 31/07/2025, 21:19:02
class Solution {
    public int climbStairs(int n) {
        if(n==1)
        {
            return 1;
        }
        if(n==2)
        {
            return 2;
        }
        int p1=1,p2=2;
        for(int i=3;i<=n;i++)
        {
            int c=p1+p2;
            p1=p2;
            p2=c;
        }
        return p2;
        
    }
}